### PR TITLE
📚 Allow multiple pages in typst export

### PR DESCRIPTION
**Pull request by rowanc1**
This allows for multi-page exports for typst

### CODE REVIEW

File: packages/myst-cli/src/build/site/manifest.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export async function resolvePageExports(session: ISession, file: string) {
         ExportFormats.docx,
         ExportFormats.pdf,
         ExportFormats.tex,
+        ExportFormats.typst,
```
Comment (fwkoch) [inline packages/myst-cli/src/build/site/manifest.ts:32]: Add to the site 👍

File: packages/jtex/src/typst/imports.ts [hunk 2]
```diff
@@ -3,18 +3,18 @@ import type { TypstTemplateImports } from '../types.js';
 import { writeFileToFolder } from 'myst-cli-utils';
 
 export function renderTypstImports(
-  output: string,
+  output: string | false,
```
Comment (fwkoch) [inline packages/jtex/src/typst/imports.ts:6]: Ah, this stops writing `myst.typ` multiple times for multi-page export. Shouldn't be `false` in other cases.

File: packages/myst-cli/src/build/site/manifest.ts (hunk 1 above)
Comment (rowanc1) [inline packages/myst-cli/src/build/site/manifest.ts:32]: @fwkoch I am not actually sure we should add this, can you confirm?

File: packages/myst-cli/src/build/typst.ts [hunk 3]
```diff
@@ -261,12 +262,13 @@ export async function localArticleToTypstTemplated(
     const includeFileBases = results.map((result, ind) => {
       const base = `${name}-${content[ind]?.slug ?? ind}${ext}`;
       const includeFile = path.format({ dir, ext, base });
-      writeFileToFolder(includeFile, result.value);
+      const exports = renderTypstImports(false, collected);
+      writeFileToFolder(includeFile, `${versionString}\n\n${exports}\n\n${result.value}`);
```
Comment (rowanc1) [inline packages/myst-cli/src/build/typst.ts:266]: Typst exports on different pages need to also have imports in each page.

File: packages/jtex/src/typst/imports.ts [hunk 4]
```diff
@@ -3,18 +3,18 @@ import type { TypstTemplateImports } from '../types.js';
...
   const importStatements: string[] = [];
-  if (macros && macros.length > 0) {
+  const hasMacros = macros && macros.length > 0;
+  if (hasMacros) importStatements.push('#import "myst.typ": *');
+  if (output && hasMacros) {
     const mystTypst = path.join(path.dirname(output), 'myst.typ');
-    importStatements.push('#import "myst.typ": *');
     writeFileToFolder(mystTypst, macros.join('\n\n'));
   }
-  importStatements.push('#set math.equation(numbering: "(1)")');
```
Comment (fwkoch) [inline packages/jtex/src/typst/imports.ts:17]: Was this moved to the template?

File: packages/myst-cli/src/build/typst.ts [hunk 5]
```diff
@@ -261,12 +262,13 @@ export async function localArticleToTypstTemplated(
     const includeFileBases = results.map((result, ind) => {
       const base = `${name}-${content[ind]?.slug ?? ind}${ext}`;
       const includeFile = path.format({ dir, ext, base });
-      writeFileToFolder(includeFile, result.value);
+      const exports = renderTypstImports(false, collected);
```
Comment (fwkoch) [inline packages/myst-cli/src/build/typst.ts:265]: Ugh, a bit tedious we have to add these here, but they get added to the main page during render... but makes sense.

**Review by fwkoch — APPROVED** [review]

File: packages/jtex/src/typst/imports.ts (hunk 4 above)
Comment (rowanc1) [reply packages/jtex/src/typst/imports.ts:17]: Yes!

File: packages/myst-cli/src/build/typst.ts (hunk 5 above)
Comment (rowanc1) [reply packages/myst-cli/src/build/typst.ts:265]: Yeah, I was a bit suprised.

**Pull request merged**